test/distributed/org/apache/cassandra/distributed/test/ReadRepairTest.java

```diff
@@ -250,10 +402,53 @@ public void readRepairRTRangeMovementTest() throws Throwable
-
-    private void consume(Iterator<Object[]> it)
+    /**
+     * Range queries before CASSANDRA-11427 will trigger read repairs for puregable tombstones on hosts that already
+     * compacted given tombstones. This will result in constant transfer and compaction actions sourced by few nodes
+     * seeding purgeable tombstones and triggered e.g. by periodical jobs scanning data range wise.
+     * <p>
+     * See CASSANDRA-11427.
+     * <p>
+     * Migrated from Python dtest read_repair_test.py:TestReadRepair.test_gcable_tombstone_resurrection_on_range_slice_query()
+     */
+    @Test
+    public void testGCableTombstoneResurrectionOnRangeSliceQuery() throws Throwable
     {
-        while (it.hasNext())
-            it.next();
+        try (Cluster cluster = init(Cluster.create(2)))
+        {
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.t (k int, c int, PRIMARY KEY(k, c)) " +
+                                              "WITH gc_grace_seconds=0 AND compaction = " +
+                                              "{'class': 'SizeTieredCompactionStrategy', 'enabled': 'false'}"));
+
+            // insert some data
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.t(k, c) VALUES (0, 0)"), ALL);
+            cluster.coordinator(1).execute(withKeyspace("INSERT INTO %s.t(k, c) VALUES (1, 1)"), ALL);
+
+            // create partition tombstones in all nodes for both existent and not existent partitions
+            cluster.coordinator(1).execute(withKeyspace("DELETE FROM %s.t WHERE k=0"), ALL); // exists
+            cluster.coordinator(1).execute(withKeyspace("DELETE FROM %s.t WHERE k=2"), ALL); // doesn't exist
+
+            // create row tombstones in all nodes for both existent and not existent rows
+            cluster.coordinator(1).execute(withKeyspace("DELETE FROM %s.t WHERE k=1 AND c=1"), ALL); // exists
+            cluster.coordinator(1).execute(withKeyspace("DELETE FROM %s.t WHERE k=3 AND c=1"), ALL); // doesn't exist
+
+            // flush single sstable with tombstones
+            cluster.get(1).flush(KEYSPACE);
+            cluster.get(2).flush(KEYSPACE);
+
+            // purge tombstones from node2 with compaction (gc_grace_seconds=0)
+            cluster.get(2).forceCompact(KEYSPACE, "t");
+
+            // execute range slice query, which should not trigger read-repair for purged TS
+            final UUID sessionId = UUIDGen.getTimeUUID();
+            cluster.coordinator(1).executeWithTracing(sessionId, withKeyspace("SELECT * FROM %s.t"), ALL);
+            List<TracingUtil.TraceEntry> traces = TracingUtil.getTrace(cluster, sessionId);
+            traces.forEach(e -> {
+                String activity = e.activity;
+                String errorMsg = "Found unexpected read-repair messages";
+                assertFalse(errorMsg, activity.contains(READ_REPAIR_REQ.toString()));
+                assertFalse(errorMsg, activity.contains(READ_REPAIR_RSP.toString()));
+            });
```

Comment:
Similar comment as [here](https://github.com/apache/cassandra/pull/708/files?file-filters%5B%5D=.java#r489761145)

Reply:
I have replaced it by a RR metrics check.